Doc/tools/check-warnings.py

```diff
@@ -94,9 +235,10 @@ def fail_if_improved(
 def main() -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "--check-and-annotate",
-        help="Comma-separated list of files to check, "
-        "and annotate those with warnings on GitHub Actions",
+        "--check-and-annotate-changed",
+        nargs=2,
```

Comment:
The `--help` shows:
```
  --check-and-annotate-changed CHECK_AND_ANNOTATE_CHANGED CHECK_AND_ANNOTATE_CHANGED
                        Annotate lines changed between two refs with warnings on GitHub Actions
```

Shall we add some sort of nicer display name?

```suggestion
        nargs=2,
        metavar="commit",
```

```
  --check-and-annotate-changed commit commit
                        Annotate lines changed between two refs with warnings on GitHub Actions
```

Reply:
Yeah, though it's actually any Git ref (not necessary, and probably not even usually a commit), the lower-case `commit` is hard to parse as a placeholder for me, and we should be clear on the actual semantics of the two different refs, the first being the base ref being compared against and the second being the head ref to compare (which can be done by passing a tuple of strings to `metavar`).

I did have to tweak this a bit to implement your following suggestion (due to `nargs="*"`), but it ends up looking alright, I think:

```text
  --annotate-diff [BASE_REF [HEAD_REF ...]]
                        Add GitHub Actions annotations on the diff for warnings on lines changed between the given refs (main and HEAD, by default)
```